spec/lib/saml_idp/request_spec.rb

```diff
@@ -1,32 +1,46 @@
 require 'spec_helper'
 module SamlIdp
   describe Request do
-    let(:raw_request) { "<samlp:AuthnRequest AssertionConsumerServiceURL='http://localhost:3000/saml/consume' Destination='http://localhost:1337/saml/auth' ID='_af43d1a0-e111-0130-661a-3c0754403fdb' IssueInstant='2013-08-06T22:01:35Z' Version='2.0' xmlns:samlp='urn:oasis:names:tc:SAML:2.0:protocol'><saml:Issuer xmlns:saml='urn:oasis:names:tc:SAML:2.0:assertion'>localhost:3000</saml:Issuer><samlp:NameIDPolicy AllowCreate='true' Format='urn:oasis:names:tc:SAML:1.1:nameid-format:emailAddress' xmlns:samlp='urn:oasis:names:tc:SAML:2.0:protocol'/></samlp:AuthnRequest>" }
-    subject { described_class.new raw_request }
+    describe "with an inflated and valid SAML request" do
+      let(:raw_request) { "<samlp:AuthnRequest AssertionConsumerServiceURL='http://localhost:3000/saml/consume' Destination='http://localhost:1337/saml/auth' ID='_af43d1a0-e111-0130-661a-3c0754403fdb' IssueInstant='2013-08-06T22:01:35Z' Version='2.0' xmlns:samlp='urn:oasis:names:tc:SAML:2.0:protocol'><saml:Issuer xmlns:saml='urn:oasis:names:tc:SAML:2.0:assertion'>localhost:3000</saml:Issuer><samlp:NameIDPolicy AllowCreate='true' Format='urn:oasis:names:tc:SAML:1.1:nameid-format:emailAddress' xmlns:samlp='urn:oasis:names:tc:SAML:2.0:protocol'/></samlp:AuthnRequest>" }
+      subject { described_class.new raw_request }
 
-    it "has a valid request_id" do
-      subject.request_id.should == "_af43d1a0-e111-0130-661a-3c0754403fdb"
-    end
+      it "has a valid request_id" do
+        subject.request_id.should == "_af43d1a0-e111-0130-661a-3c0754403fdb"
+      end
 
-    it "has a valid acs_url" do
-      subject.acs_url.should == "http://localhost:3000/saml/consume"
-    end
+      it "has a valid acs_url" do
+        subject.acs_url.should == "http://localhost:3000/saml/consume"
+      end
 
-    it "has a valid service_provider" do
-      subject.service_provider.should be_a ServiceProvider
-    end
+      it "has a valid service_provider" do
+        subject.service_provider.should be_a ServiceProvider
+      end
 
-    it "has a valid service_provider" do
-      subject.service_provider.should be_truthy
-    end
+      it "has a valid service_provider" do
+        subject.service_provider.should be_truthy
+      end
+
+      it "has a valid issuer" do
+        subject.issuer.should == "localhost:3000"
+      end
 
-    it "has a valid issuer" do
-      subject.issuer.should == "localhost:3000"
+      it "has a valid valid_signature" do
+        subject.valid_signature?.should be_truthy
+      end
     end
 
-    it "has a valid valid_signature" do
-      subject.valid_signature?.should be_truthy
+    describe "with an inflated but invalid SAML request" do
+      it "returns an empty string when " do
+        described_class.new "<samlp:AuthnRequest></samlp:AuthnRequest>"
+        subject.acs_url == ""
```

Comment:
Could be good to have a few of those other checks ran too to see that the rest of the class doesn't fall over in the absence of the authnrequest